File: backend-model/src/main/java/outcobra/server/model/dto/SubjectDto.java
package outcobra.server.model.dto;

import outcobra.server.model.Subject;
import outcobra.server.model.interfaces.MappableDto;
import outcobra.server.model.interfaces.Mapper;

public class SubjectDto implements MappableDto<SubjectDto, Subject> {
    @Override
    public Mapper<SubjectDto, Subject> getMapper() {
        throw new UnsupportedOperationException("not implemented");
    }

    @Override
    public Subject toEntity() {
        throw new UnsupportedOperationException("not implemented");
    }
}

File: backend/src/main/kotlin/outcobra/server/model/dto/SubjectDto.kt
package outcobra.server.model.dto

import outcobra.server.model.domain.User
import outcobra.server.model.interfaces.OutcobraDto
import outcobra.server.model.interfaces.ParentLink

data class SubjectDto(val id: Long = 0,
                      val schoolClassSemesterDto: List<SchoolClassSemesterDto> = listOf(),
                      val name: String = "",
                      val color: ColorDto = ColorDto(),
                      val userId: Long = 0,
                      val teacherId: Long? = 0) : OutcobraDto {

    override val identifier: Long get() = id
    override val parentLink: ParentLink
        get() = ParentLink.make(userId, User::class.java)
}
